File: src/wrappers/python/examples/capture_example.py
#!/usr/bin/env python3
"""
ITSCAM SDK Python Example - Binary TCP client

Connects to the camera on port 60000, optionally authenticates, requests
a snapshot, and waits briefly for trigger frames.

Usage:
    python3 capture_example.py <camera_ip> [password]

Copyright (c) 2026 Pumatronix
"""
from __future__ import annotations

import sys
import time
from pathlib import Path

sys.path.insert(0, str(Path(__file__).parent.parent))

from itscam import ItscamClient


def main() -> int:
    if len(sys.argv) < 2:
        print(f"Usage: {sys.argv[0]} <camera_ip> [password]")
        return 1

    host = sys.argv[1]
    password = sys.argv[2] if len(sys.argv) > 2 else ""

    snapshot_count = 0

    def on_snapshot(result) -> None:
        nonlocal snapshot_count
        snapshot_count += 1
        print(
            f"  snapshot callback #{snapshot_count}: "
            f"{len(result.jpeg)} bytes"
        )

    print(f"Connecting to {host}:60000 ...")
    with ItscamClient(host, timeout_ms=10000) as client:
        if password:
            client.authenticate(password)
            print("Authenticated.")

        client.on_snapshot_image(on_snapshot)
        client.subscribe_captures()

        results = client.capture_snapshot()
        print(f"capture_snapshot returned {len(results)} frame(s)")
        for i, result in enumerate(results):
            print(f"  frame {i + 1}: {len(result.jpeg)} bytes")

        print("Waiting briefly for trigger frames ...")
        time.sleep(2)

    print(f"Done ({snapshot_count} snapshot callback(s)).")
    return 0


if __name__ == "__main__":
    sys.exit(main())

File: src/wrappers/nodejs/examples/capture-example.js
#!/usr/bin/env node
/*
 * SPDX-License-Identifier: Proprietary
 * Copyright (c) 2026 Pumatronix
 *
 * Binary client capture example.  Mirrors capture_example.py /
 * capture_example.go / BinaryCaptureExample/Program.cs.
 *
 * Usage:
 *   node capture-example.js <camera_ip> [password]
 */
'use strict';

const path = require('path');
const { setTimeout: sleep } = require('timers/promises');
const { ItscamClient, getNativeLibraryVersion } =
    require(path.join(__dirname, '..', 'src'));

async function main() {
    if (process.argv.length < 3) {
        console.error('Usage: capture-example.js <camera_ip> [password]');
        process.exit(1);
    }
    const host = process.argv[2];
    const password = process.argv[3] || null;

    console.log('ITSCAM SDK', getNativeLibraryVersion());

    let snapshotCount = 0;
    const camera = new ItscamClient();
    try {
        console.log(`Connecting to ${host}:60000 ...`);
        camera.connect(host, 60000, 10000, { enabled: true });

        if (password) {
            camera.authenticate(password, 10000);
            console.log('Authenticated.');
        }

        camera.onSnapshotImage((result) => {
            snapshotCount++;
            console.log(
                `  snapshot callback #${snapshotCount}: ${result.jpeg.length} bytes`);
        });

        camera.subscribeCaptures(undefined, 10000);

        const frames = camera.captureSnapshot(15000);
        console.log(`captureSnapshot returned ${frames.length} frame(s)`);
        frames.forEach((r, i) => {
            console.log(`  frame ${i + 1}: ${r.jpeg.length} bytes, plates=${JSON.stringify(r.plates)}`);
        });

        console.log('Waiting briefly for trigger frames ...');
        await sleep(2000);
    } finally {
        camera.close();
    }

    console.log(`Done (${snapshotCount} snapshot callback(s)).`);
}

main().catch((err) => {
    console.error(err);
    process.exit(1);
});
